### ISSUE TITLE
Checkboxes track checked state based on position

### ISSUE THREAD

**Issue by Nathan-Schwartz** [ConciergeAuctions/react-mui-table#32]
This causes issues when sorting, searching or deleting after checking a box. The rows switch positions, but the original row stays checked.

The quick and temporary solution would be to uncheck everything when doing any of these actions.

**Comment by brad-decker** [comment]
I would expect that checkboxes be unchecked when filtering/ordering/searching/deleting 

@iamdavid0091 UX opinion?

**Comment by Nathan-Schwartz** [comment]
@brad-decker Actions all have their own callback so unselecting all for each individual interaction could be a pain. I think a reasonable solution would be to unselect all when the `items` prop changes. Thoughts?

**Comment by iamdavid0091** [comment]
@brad-decker I would expect the checkboxes to stay checked when sorting. When filtering or searching, the checkboxes should be unchecked.

**Comment by brad-decker** [comment]
@iamdavid0091 with sorting the checkbox might go to another page - i would find it confusing if i had checked something and it went to page 5 and i no longer knew i had it checked .

**Comment by brad-decker** [comment]
@Nathan-Schwartz its worth a shot to see if it doesn't introduce any unexpected behavior

**Comment by Nathan-Schwartz** [comment]
@brad-decker Okay, I'll open a new PR to fix this bug and then rebase my existing PR once it is merged 👍

**Comment by iamdavid0091** [comment]
@brad-decker @Nathan-Schwartz a good example is google analytics. Check out how they handle checked/unchecked states—I think they have the same implementation of what I described earlier.

**Comment by brad-decker** [comment]
@iamdavid0091 - analytics workflow is broken, at best. 

https://analytics.google.com/analytics/web/#report/content-pages/a25446921w49461106p136018215/

if you check 'upcoming auctions' then sort by pageviews reverse upcoming auctions goes out of view. The only indicator that something is checked is the '[-]' button in the top left of the table. The only action i can take when i check a box is 'plot' them but if the checked item is out of view i can't even do that. 

I really believe that if a checked item goes out of view it should be unchecked. The only time i think its cool to check items that aren't in view is in a classic gmail style where you select all, click delete, then it asks you if you want to delete the rest and thats not even really checking items on another page.

**Comment by Nathan-Schwartz** [comment]
@iamdavid0091 @brad-decker One solution could be to allow items to stay checked if they remain in view while sorting / searching and remove the ones that are no longer on the page. I think the implementation for this is more complex and I'm not sure if it would be intuitive for users at first. I think it would be safe and convenient once people got used to it.

**Comment by iamdavid0091** [comment]
@brad-decker I don't agree. If the use case is that I want to plot a graph of the page w/ the most amount of pageviews and of the page w/ the least amount of pageviews, I would want the my checked state of /upcoming-auctions to be persistent so that when I sort by the inverse to get the page w/ the least amount of pageviews, I'm able to check the new row w/o losing my checked state of /upcoming-auctions.

**Comment by brad-decker** [comment]
@iamdavid0091 i can see that use case. I wasn't thinking i had to have at least two items selected :P. I don't think you'd ever want to delete something that was checked on another page though, right? I think thats potentially dangerous. We don't have any reason for multiselect currently other than deletion.

**Comment by iamdavid0091** [comment]
If we go with a design where for the collections object we have a properties table with checkboxes, onSelection of multiple checkboxes, we have the FAB button go from a disabled state to an enabled state, I can see how the use case described in my previous comment would also come into play.

**Comment by brad-decker** [comment]
adding properties to the table is being handled, currently by a modal with an entirely different list implementation. If we need to revisit this in the future when there is a use case for keeping the state of checkboxes checked when they are off page we can revisit that. for now i think our current use case is met.

What @iamdavid0091 says is true. There may be a time in the future where it would make sense. Want to create a new issue to track that as a future enhancement @Nathan-Schwartz ?
Reference this issue for the convo

**Comment by Nathan-Schwartz** [comment]
Sure, I'll tag @iamdavid0091 in the issue to make sure I'm not leaving stuff out 👍
